src/App.js

```diff
@@ -111,47 +117,111 @@ class App extends Component {
       await this.initialFetch(session.idToken.payload.sub);
     } catch (e) {
       if (e === "No current user") {
-        this.setCloseLoading();
+        const result = await fetchSanitySchemas();
+        if (result.success) {
+          this.setState({ sanitySchemas: result.sanitySchemas }, () =>
+            this.setCloseLoading()
+          );
+        } else {
+          // TODO: If success === false, redirect to a page indicating a app error and to try again later (not the 404 not found page)
+          this.setCloseLoading();
+        }
       }
       if (e !== "No current user") {
         errorToast(e, this.state.user);
+        this.setCloseLoading();
       }
-      this.setState({ loading: false });
     }
     this.setState({ isAuthenticating: false });
   }
 
   initialFetch = async (username) => {
-    try {
-      const user = await API.get("pareto", `/users/${username}`);
-      if (user.length > 0) {
-        this.props.getUser(user[0]);
-        this.setState({ user: user[0] });
-        if (user[0].defaultLanguage) {
+    let newState = {};
+    const path = this.props.location.pathname;
+
+    // Set up variables to enable fetching only the data needed for your current app view
+    const [context, training, arena] = [
+      path.includes("context-builder"),
+      path.includes("training"),
+      !path.includes("context-builder") && !path.includes("training"),
+    ];
+    const firstFetch = [];
+    const secondFetch = [];
+
+    const socketOptions = { updateState: false };
+
+    const user = await fetchUser(username);
+    if (user.length > 0) {
+      const currentUser = user[0];
+      try {
+        this.props.getUser(currentUser);
+        newState.user = currentUser;
+        if (currentUser.defaultLanguage) {
           const language = availableLanguages.find(
-            (x) => x.code === user[0].defaultLanguage
+            (x) => x.code === currentUser.defaultLanguage
           );
-          this.setState({
-            chosenLanguage: language || "en",
-          });
-          I18n.setLanguage(user[0].defaultLanguage);
+          newState.chosenLanguage = language || "en";
+          I18n.setLanguage(currentUser.defaultLanguage);
         }
-
-        // if (user[0].learningPurchase === true || user[0].instructor === true) {
-        await this.fetchStarterKitExperience(user[0].id);
-        await this.fetchStarterKitSanity();
-        // }
-
-        await this.fetchCoaches(user[0].id);
-        if (user[0].instructor === true) {
-          await this.fetchCoachingRoster(user[0].id);
+        socketOptions.user = currentUser;
+
+        // Sort fetching functions according to whether they should happen before or after the loading overlay goes away
+        if (context) {
```

Comment:
This is amazing. Soon, we can remove the rendering overlay altogether and just have skeleton UI elements that get slowly revealed! This will definitely improve our lighthouse score, I would imagine!